packages/shorebird_code_push_client/lib/src/code_push_client.dart

```diff
@@ -431,9 +439,11 @@ class CodePushClient {
   void close() => _httpClient.close();
 
   CodePushException _parseErrorResponse(int statusCode, String response) {
-    final exceptionBuilder = statusCode == HttpStatus.conflict
-        ? CodePushConflictException.new
-        : CodePushException.new;
+    final exceptionBuilder = switch (statusCode) {
+      HttpStatus.conflict => CodePushConflictException.new,
+      HttpStatus.notFound => CodePushNotFoundException.new,
+      _ => CodePushException.new,
```

Comment:
nice!

Reply:
Pattern matching is so nice. Thrilled to have a reason to use it :)